test/system/users_test.rb

```diff
@@ -0,0 +1,16 @@
+require "application_system_test_case"
+
+class UsersTest < ApplicationSystemTestCase
+  # test "visiting the index" do
+  #   visit users_url
+  #
+  #   assert_selector "h1", text: "Users"
+  # end
+  test "preserve return_to while switching from login to signup" do
+    visit '/login?return_to=/subscribe/multiple/tag/tag1,tag2'
+    click_on "sign up"
+    assert_text "Create a password"
+    assert_text "Confirmation"
+    assert_text "Bio"
```

Comment:
I don't think this test does what you mean or what I understand from the title. You want to test that a user is returned to a URL after they try to login and it appears they don't have an account, so after they sign up, they are returned to the URL they were trying to access?

Reply:
I think you have not understood the functionality properly. This is the continued work of #5559.
The test ensures that when the user clicks on the ``sign up `` button on the login page with url `/login?return_to=x` ,then they get redirected to the signup page with the url '/signup?return_to=x`.

I am sorry for the inconvenience.

Reply:
Thanks for the input. I have modified the tests according to your suggestions.